core/templates/dev/head/pages/topic_editor/TopicEditorStateService.js

```diff
@@ -321,6 +323,21 @@ oppia.factory('TopicEditorStateService', [
           commitMessage, UndoRedoService.getCommittableChangeList()).then(
           function(topicBackendObject) {
             _updateTopic(topicBackendObject);
+            var changeList = UndoRedoService.getCommittableChangeList();
+            for (var i = 0; i < changeList.length; i++) {
+              if (changeList[i].property_name === 'canonical_story_ids') {
+                if (changeList[i].new_value.length <
+                    changeList[i].old_value.length) {
```

Comment:
Technically you should check that the new value length is 1 less than the old value, and throw an error if it isn't. Don't assume.